File: wrappers/dotnet/indy-sdk-dotnet-test/Wrapper/AgentTests/AgentListenTest.cs
using Indy.Sdk.Dotnet.Wrapper;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace Indy.Sdk.Dotnet.Test.Wrapper.AgentTests
{
    [TestClass]
    public class AgentListenTest : AgentIntegrationTestBase
    {
        [TestMethod]
        public void TestAgentListenWorks()
        {
            var endpoint = "127.0.0.1:9607";

            var didJson = "{\"seed\":\"sovrin_agent_connect_works_for_a\"}";
       
            var myDidResult = Signus.CreateAndStoreMyDidAsync(_wallet, didJson).Result;

            var identityJson = string.Format("{{\"did\":\"{0}\", \"pk\":\"{1}\", \"verkey\":\"{2}\", \"endpoint\":\"{3}\"}}",
                    myDidResult.Did, myDidResult.Pk, myDidResult.VerKey, endpoint);
            Signus.StoreTheirDidAsync(_wallet, identityJson).Wait();

            Agent.AgentListenAsync(endpoint, _incomingConnectionObserver).Wait();
        }
    }
}

File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/agent/AgentListenTest.java
package org.hyperledger.indy.sdk.agent;

import org.hyperledger.indy.sdk.signus.Signus;
import org.hyperledger.indy.sdk.signus.SignusJSONParameters;
import org.hyperledger.indy.sdk.signus.SignusResults;
import org.junit.Test;


public class AgentListenTest extends AgentIntegrationTest {

	@Test
	public void testAgentListenWorksForAllDataInWalletPresent() throws Exception {
		String endpoint = "127.0.0.1:9607";

		SignusResults.CreateAndStoreMyDidResult myDid = Signus.createAndStoreMyDid(wallet, "{}").get();

		String identityJson = String.format(AGENT_IDENTITY_JSON_TEMPLATE, myDid.getDid(), myDid.getPk(), myDid.getVerkey(), endpoint);
		Signus.storeTheirDid(wallet, identityJson).get();

		Agent.agentListen(endpoint, incomingConnectionObserver).get();
	}
}